### ISSUE TITLE
[a11y]: HeaderMenuButton does not have a visible label with keyboard navigation

### ISSUE THREAD

**Issue by Alienah** [carbon-design-system/carbon#20029]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.78.0

### React version

v19.1.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The following accessibility checker rule indicates that an input (in this case a button, the `HeaderMenuButton`) must have a visible label. Although there is currently a `title`, which appears on the mouse hover, this is not the case for keyboard navigation.

<img width="895" height="816" alt="Image" src="https://github.com/user-attachments/assets/1e1957de-29e1-4bf3-94e6-498e9f75f122" />

There should be a visible label for both mouse and keyboard navigation, as it does for example in the `IconButton`, which uses a tooltip.

### WCAG 2.1 Violation

3.3.2 Labels or Instructions

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-navigation-and-actions

### Steps to reproduce

* Navigate with the keyboard through the header and see that the HeaderMenuButton does not show the label "Open menu" when focus
* See that the rest of the icons buttons, like Search, Notifications and App switcher work as expected showing their labels when keyboard focus

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
Hey @Alienah Search, Notifications and App switcher are icon only Buttons and label is required on them.
But HeaderMenuButton does not need to render a label on hover 

Accessibility Checker throws no error as well 
<img width="1702" height="796" alt="Image" src="https://github.com/user-attachments/assets/8d4e10c6-19f2-4e38-9680-1ef2c0aceb9a" />

**Comment by Alienah** [comment]
@2nikhiltom Why doesn't it need a label if it's an icon button too? Anyway, the issue appears in _Needs review_, not in _Violations_, maybe that's why you don't see it?
